deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/security/EnvStageBodyExtractor.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2024 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.teletraan.security;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.pinterest.deployservice.ServiceContext;
+import com.pinterest.deployservice.bean.AgentBean;
+import com.pinterest.deployservice.bean.DeployBean;
+import com.pinterest.deployservice.bean.EnvironBean;
+import com.pinterest.deployservice.bean.HotfixBean;
+import com.pinterest.deployservice.dao.EnvironDAO;
+import com.pinterest.teletraan.universal.security.AuthZResourceExtractor;
+import com.pinterest.teletraan.universal.security.bean.AuthZResource;
+import java.io.InputStream;
+import javax.ws.rs.container.ContainerRequestContext;
+
+public class EnvStageBodyExtractor implements AuthZResourceExtractor {
+    private final EnvironDAO environDAO;
+
+    public EnvStageBodyExtractor(ServiceContext context) {
+        this.environDAO = context.getEnvironDAO();
+    }
+
+    @Override
+    public AuthZResource extractResource(ContainerRequestContext requestContext, Class<?> beanClass)
+            throws ExtractionException {
+        return extractResource(requestContext, beanClass, false);
+    }
+
+    @Override
+    public AuthZResource extractResource(ContainerRequestContext requestContext)
+            throws ExtractionException {
+        return extractResource(requestContext, Object.class, true);
+    }
+
+    AuthZResource extractResource(
```

Comment:
This method is hard to read. Please refactor it following clean code and SOLID principles.

An example of refactoring I added in the comment for the [TokenAuthenticationFactory.java](https://github.com/pinterest/teletraan/pull/1487/files#diff-889649966b3c388c596bd54ddf09c977d9e0a508bc4efad938a824a4ad1c8a69) class.